oap-ape/ape-java/ape-common/src/main/java/com/intel/ape/service/netty/NettyMessage.java

```diff
@@ -0,0 +1,491 @@
+        }
+        throw new IOException(t);
+    }
+
+    // ------------------------------------------------------------------------
+    // Client requests
+    // ------------------------------------------------------------------------
+    public static class ParquetReaderInitRequest extends NettyMessage {
+        public static final byte ID = 0;
+
+        private ParquetReaderInitParams params;
+
+        public ParquetReaderInitRequest(ParquetReaderInitParams params) {
+            this.params = params;
+        }
+
+        @Override
+        void write(ChannelOutboundInvoker out, ChannelPromise promise, ByteBufAllocator allocator)
+                throws IOException {
+            final ByteBuf result = allocateBuffer(allocator, ID);
+
+            try (ObjectOutputStream oos = new ObjectOutputStream(new ByteBufOutputStream(result))) {
+                oos.writeObject(params);
+
+                // Update frame length...
+                result.setInt(0, result.readableBytes());
+                out.write(result, promise);
+            }
+            catch (Throwable t) {
+                handleException(result, t);
+            }
+        }
+
+        static ParquetReaderInitRequest readFrom(ByteBuf buffer) throws Exception {
+            try (ObjectInputStream ois = new ObjectInputStream(new ByteBufInputStream(buffer))) {
+                Object obj = ois.readObject();
+
+                if (!(obj instanceof ParquetReaderInitParams)) {
+                    throw new ClassCastException(
+                            "Read object expected to be of type ReaderInitParams, " +
+                            "actual type is " + obj.getClass() + ".");
+                } else {
+                    return new ParquetReaderInitRequest((ParquetReaderInitParams) obj);
+                }
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "ReaderInitRequest{" +
+                    "params=" + params +
+                    '}';
+        }
+    }
+
+    public static class ReadBatchRequest extends NettyMessage {
+        public static final byte ID = 1;
+
+        private final int batchCount;
+
+        public ReadBatchRequest(int batchCount) {
+            this.batchCount = batchCount;
+        }
+
+        @Override
+        void write(ChannelOutboundInvoker out, ChannelPromise promise, ByteBufAllocator allocator)
+                throws IOException {
+            Consumer<ByteBuf> consumer = (bb) -> {
+                bb.writeInt(batchCount);
+            };
+
+            writeToChannel(out, promise, allocator, consumer, ID, 4);
+        }
+
+        static ReadBatchRequest readFrom(ByteBuf buffer) throws IOException {
+            return new ReadBatchRequest(buffer.readInt());
+        }
+
+        public int getBatchCount() {
+            return batchCount;
+        }
+    }
+
+    public static class HasNextRequest extends NettyMessage {
+        public static final byte ID = 2;

```

Comment:
I think we could define some static variables like `int MESSAGE_ID_HAS_NEXT_REQUEST = 2`, it will be more clear.

Reply:
Will be added in another pull request.